src/sardana/pool/poolacquisition.py

```diff
@@ -796,7 +796,7 @@ def action_loop(self):
                 if is_value_error(value):
                     self.error("Loop final read value error for: %s" %
                                acquirable.name)
-                    acquirable.put_value(value)
+                    acquirable.append_value_buffer(None, self.index)
```

Comment:
Since the `append_value_buffer` apart of sending the new value via the `Data` attribute also sets the `Value` attribute then your proposal raises an exception in the `on_ct_changed` method of the CTExpChannel device. I propose to append the value itself in both cases: error and not error. This way the error message is transferred to the clients, as it was done previously with the `put_value` call, but this way we also fill the buffer with a placeholder - `SardanaValue` with the error information. Please test and change the implementation accordingly.